src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs

```diff
@@ -72,6 +72,7 @@ public override IList<CustomAttributeData> GetCustomAttributesData()
 
         // GetDefaultMembers
         // This will return a MemberInfo that has been marked with the [DefaultMemberAttribute]
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
```

Comment:
Does the attribute have any effect here when the type is sealed?

Reply:
The attribute is on the "this" - which is a special case - it annotates the type this RuntimeType represents. So for example calling `typeof<MyType>.GetDefaultMembers()` would end up (due to this attribute) marking the entire `MyType`.

Reply:
Sure but as this is C# code there won't be any call (IL reference) to this method

Reply:
Note that the same annotation is on `Type.GetDefaultMembers()` - the current data flow implementation in the linker assumes that the entire inheritance hierarchy for a given method has the same annotations (We're missing the validation in the linker for now https://github.com/mono/linker/issues/1028).

Other than the above rule, this also means that when analyzing the method body of this method the linker can make assumptions about the `this` and won't generate warnings...